homeassistant/components/websocket_api/auth.py

```diff
@@ -83,7 +87,14 @@ async def async_handle(self, msg: dict[str, str]) -> ActiveConnection:
                 msg["access_token"]
             )
             if refresh_token is not None:
-                return await self._async_finish_auth(refresh_token.user, refresh_token)
+                conn = await self._async_finish_auth(refresh_token.user, refresh_token)
+                unregister_token_cb = (
+                    self._hass.auth.async_register_revoke_token_callback(
+                        refresh_token.id, self._cancel_ws
+                    )
+                )
+
+                return conn, unregister_token_cb
```

Comment:
No need to return it. You can add it to the subscriptions dictionary on the connection and it will be called when the connection closes:

```suggestion
                conn.subscriptions['auth'] = (
                    self._hass.auth.async_register_revoke_token_callback(
                        refresh_token.id, self._cancel_ws
                    )
                )

                return conn
```

Reply:
Ah even better :)